site/hooks/useElementWidth/index.ts

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useCallback, useState, RefObject } from "react";
+
+export const useElementWidth = (elementRef: RefObject<HTMLElement>) => {
+  const [width, setWidth] = useState<number>(0);
+
+  const updateWidth = useCallback(() => {
+    if (elementRef && elementRef.current) {
+      const { width } = elementRef.current.getBoundingClientRect();
+      setWidth(width);
+    }
+  }, [elementRef]);
+
+  const firstUpdateWidth = useCallback(() => {
+    if (elementRef && elementRef.current) {
+      let { width } = elementRef.current.getBoundingClientRect();
+      width =
+        width -
+        parseInt(
+          window
+            .getComputedStyle(elementRef.current)
+            .getPropertyValue("border-left-width")
+        ) -
+        parseInt(
+          window
+            .getComputedStyle(elementRef.current)
+            .getPropertyValue("border-right-width")
+        );
+
+      setWidth(width);
+    }
+  }, [elementRef]);
+
+  useEffect(() => {
+    firstUpdateWidth();
+    window.addEventListener("resize", updateWidth);
+    return () => {
+      window.removeEventListener("resize", updateWidth);
+    };
+  }, [updateWidth]);
+
+  return [width];
```

Comment:
would be neat if you also manage the ref inside this hook, so the parent doesnt need to pass anything in
```suggestion
  return [width, ref];
```

```
const [articleWidth, articleRef] = useElementWidth()
```

Reply:
The hook is used for the infinity slider too.
Will leave it as is right now and let´s do this in a follow-up